buyer-app/buyer-app/src/state/entities/downloadData/sagas.js

```diff
@@ -0,0 +1,45 @@
+import { put, takeLatest, all, call } from "redux-saga/effects";
+import download from "downloadjs";
+
+import * as NotificationActions from "state/entities/notifications/actions";
+import { getData } from "./helpers";
+
+import * as Actions from "./actions";
+
+/**
+ * Download data
+ * @yield {[type]} [description]
+ */
+function* downloadData(action) {
+  const { dataOrder } = action.payload;
+  try {
+    const { data, error } = yield call(getData, dataOrder.id);
+    if (error) {
+      yield put(
+        NotificationActions.createNotification({
+          message: error.message,
+          status: "critical"
+        })
+      );
+    } else {
+      const fileName = dataOrder.requestedData.join("-") + ".csv";
```

Comment:
Instead of this `fileName`, I would use one that shows the order ID, since probably they'll download multiple CSVs of same type of data, and it's clearer if they can distinguish them by order id